### ISSUE TITLE
zig fmt doesn't like being disabled for inner blocks.

### ISSUE THREAD

**Issue by nmichaels** [ziglang/zig#5548]
Before running `zig fmt` on a file, it looks like this:

```
// zig fmt: off
fn doThing() !void {
    const neighbors = [_][]u32{
        [_]u32{ -1, -1 }, [_]u32{ -1, 0 }, [_]u32{ -1, 1 },
        [_]u32{  0, -1, },                 [_]u32{  0, 1 },
        [_]u32{  1, -1 }, [_]u32{  1, 0 }, [_]u32{  1, 1 },
    };
}
// zig fmt: on

fn doThing() !void {
    // zig fmt: off
    const neighbors = [_][]u32{
        [_]u32{ -1, -1 }, [_]u32{ -1, 0 }, [_]u32{ -1, 1 },
        [_]u32{  0, -1, },                 [_]u32{  0, 1 },
        [_]u32{  1, -1 }, [_]u32{  1, 0 }, [_]u32{  1, 1 },
    };
    // zig fmt: on
}

const s = struct {
    // zig fmt: off
    fn doThing() !void {
        const neighbors = [_][]u32{
            [_]u32{ -1, -1 }, [_]u32{ -1, 0 }, [_]u32{ -1, 1 },
            [_]u32{  0, -1, },                 [_]u32{  0, 1 },
            [_]u32{  1, -1 }, [_]u32{  1, 0 }, [_]u32{  1, 1 },
        };
    }
    // zig fmt: on
};
```
After, the file looks like this:

```
// zig fmt: off
fn doThing() !void {
    const neighbors = [_][]u32{
        [_]u32{ -1, -1 }, [_]u32{ -1, 0 }, [_]u32{ -1, 1 },
        [_]u32{  0, -1, },                 [_]u32{  0, 1 },
        [_]u32{  1, -1 }, [_]u32{  1, 0 }, [_]u32{  1, 1 },
    };
}
// zig fmt: on

fn doThing() !void {
    // zig fmt: off
    const neighbors = [_][]u32{
        [_]u32{ -1, -1 },                     [_]u32{ -1, 0 }, [_]u32{ -1, 1 },
        [_]u32{
            0,
            -1,
        }, [_]u32{ 0, 1 },  [_]u32{ 1, -1 },
        [_]u32{ 1, 0 },                       [_]u32{ 1, 1 },
    };
    // zig fmt: on
}

const s = struct {
    // zig fmt: off
    fn doThing() !void {
        const neighbors = [_][]u32{
            [_]u32{ -1, -1 },                                 [_]u32{ -1, 0 }, [_]u32{ -1, 1 },
            [_]u32{
                0,
                -1,
            }, [_]u32{ 0, 1 },  [_]u32{ 1, -1 },
            [_]u32{ 1, 0 },                                   [_]u32{ 1, 1 },
        };
    }
    // zig fmt: on
};
```

I don't expect it to make any changes to the `neighbors` initializer, but it seems that zig fmt can only be disabled for outer blocks.

Some bug-report speculation: Does the magic comment have to start in the first column?

**Comment by Vexu** [comment]
This doesn't seem to be documented anywhere but `zig fmt: off/on` only works at file level scope.

**Comment by justjosias** [comment]
> This doesn't seem to be documented anywhere but zig fmt: off/on only works at file level scope.

Is a change in documentation needed or rather a change in `zig fmt`? To me it seems that this should work at all scopes, otherwise you'll end up putting expressions you don't want formatted in the global scope simply because the formatting tool only can be configured there.

**Comment by ifreund** [comment]
Fixed in #7920
